src/main/java/org/threadly/litesockets/utils/MergedByteBuffers.java

```diff
@@ -0,0 +1,365 @@
+package org.threadly.litesockets.utils;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+
+/**
+ * This class is used to combine multiple ByteBuffers into 1 simplish to use interface.
+ * It provides most of the features of a single ByteBuffer, but with the ability to perform those 
+ * operations spanning many ByteBuffers.
+ * 
+ * The Idea here is to keep from having to copy around and merge ByteBuffers as much as possible. 
+ * 
+ * @author lwahlmeier
+ *
+ */
+public class MergedByteBuffers {
```

Comment:
This class is not thread safe....it seems like this class should be...coulden't one thread be adding, and another reading from it?